private-templates-service/client/src/store/currentTemplate/actions.ts

```diff
@@ -193,6 +193,15 @@ export function updateTemplate(templateID?: string, currentVersion?: string, tem
     else {
       newTemplate.template = appState.currentTemplate.templateJSON;
     }
+
+    if (sampleDataJSON) {
+      // TODO: the clientSDK, backend, and DB adapter should be refactored such that the data is a json object and not an array of objects
```

Comment:
was the original plan to make the data an array of key-value pairs ?

Reply:
I don't know. I think Grace + Seva might know. I think the data object should be changed to be a single JSON object, but I don't think that's in the scope of this PR.